.gitignore

```diff
@@ -1,10 +1,48 @@
 # Cargo build directory
-/target
+target/
+debug/
+Carge.lock
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
 
 # Nix build directory
-/result
-/result-lib
+result
+result-*
 
 # Direnv
 /.envrc
 /.direnv
+
+
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+.idea/
+
+# IntelliJ
+out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Cursive Clojure plugin
+.idea/replstate.xml
+
+# SonarLint plugin
+.idea/sonarlint/
+
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
+
+## Others
+.metadata/
+himalaya.iml
```

Comment:
What are those two last for?

Reply:
`.metadata` was due to my neovim-setup because the java-lsp seems to startup (for whatever reason) and I couldn't fix it yet. This can be probably removed since this case just comes up due to me or can I let it in for the time being?

`himalaya.iml`: I don't really know.... but it seems like that it was generated by the java-lsp again (ugh). I'll remove this

Reply:
I guess you can remove both of them so, thanks for the details :ok_hand: